crates/oxc_query/src/vertex.rs

```diff
@@ -278,6 +278,29 @@ impl<'a> Vertex<'a> {
         }
     }
 
+    pub fn function_scope_flag(&self, adapter: &Adapter<'_>) -> ScopeFlags {
+        let wanted_node_hash = match &self {
+            Vertex::ArrowFunction(data) => calculate_hash(data.arrow_expression),
+            Vertex::FnDeclaration(data) => calculate_hash(data.function),
+            _ => unreachable!(
+                "'function_scope_flag' function should only ever be called with an ArrowFunction or FnDeclaration"
+            ),
+        };
+
+        let found = adapter.semantic.nodes().iter().find(|x| {
+            let hash_of_node = match x.kind() {
+                AstKind::ArrowExpression(ae) => calculate_hash(ae),
+                AstKind::Function(fn_) => calculate_hash(fn_),
+                _ => return false,
+            };
+            hash_of_node == wanted_node_hash
```

Comment:
This fails to find the correct function on line 2, since both functions have the same hash. Maybe augmenting by also filtering for the same start and end span would work?

```
{ class X { set foo(v) { return String(v) } } }
{ x = function (v) { return String(v); } }
```

Reply:
this is fixed in #1527 to my knowledge.

Reply:
Yes, hash + span should work.